### ISSUE TITLE
Category sorting wrong if multiple categories has same after_category_id

### ISSUE THREAD

**Issue by ascheider** [shopware/platform#3596]
### PHP Version

8.2

### Shopware Version

6.4 or 6.5

### Expected behaviour

The sorting of the navigation items in storefront should be same like in the administration. 
The after_category_id should appear only once and be unique

### Actual behaviour

The sorting in the navigation is wrong if multiple categories have the same after_category_id

### How to reproduce

We have a case where the storefront navigation sorting is wrong.
The category "Test Kacheln" is not active at the bottom, but has somehow the same after_category_id like "Betriebsausstattung"
![image](https://github.com/shopware/shopware/assets/6016485/59092338-ba1e-47e2-82de-85816d746157)

![image](https://github.com/shopware/shopware/assets/6016485/87af7535-7529-4245-9245-945f9017f4e2)

So the sorting in storefront is wrong:
![image](https://github.com/shopware/shopware/assets/6016485/03d73bab-cd49-4794-b8c1-9aa3debe8d26)

Shopware sorts the categories first with: AfterSort::sort in Shopware\Core\Framework\DataAbstractionLayer\Util and then filters inactive and invisible categories. AfterSort sorts the categories wrong if after_category_id appears more then once.

**Comment by MarcelSchmaeing** [comment]
Hi,

thx for your issue we will put it into our new evaluation process and have a look at it again.

**Comment by shopwareBot** [comment]
[public] Automated response: This issue is linked to the internal issue [https://shopware.atlassian.net/browse/NEXT-37943](https://shopware.atlassian.net/browse/NEXT-37943).

 

[created from NEXT-37943, comment 489005]

**Comment by shopwareBot** [comment]
[public] Hello,

 Within our company roadmap and work capacity, we try to address each bug or improvement request but admit that not each one will be resolved. To continue our culture of honesty and openness, we are closing this issue to focus on our roadmap on behalf of all Shopware users.

 If you feel like this issue is still impacting you, please create a new issue and let us know. Thank you again for your valuable feedback.



[created from NEXT-37943, comment 549943]
